geoh5py/shared/concatenation/drillholes_group_table.py

```diff
@@ -0,0 +1,377 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from abc import ABC
+from typing import TYPE_CHECKING, Any
+from uuid import UUID
+
+import numpy as np
+
+from ..utils import to_tuple
+from .property_group import ConcatenatedPropertyGroup
+
+if TYPE_CHECKING:
+    from .concatenator import Concatenator
+    from .data import ConcatenatedData
+    from .drillhole import ConcatenatedDrillhole
+
+
+class DrillholesGroupTable(ABC):
+    """
+    A class to store the information of a PropertyGroup.
+
+    :param parent: The parent DrillholesConcatenator.
+    :param name: The name of the PropertyGroup to extract.
+    """
+
+    def __init__(self, parent: Concatenator, name: str):
+        self._association: tuple = ()
+        self._association_type: str = ""
+        self._depth_table: np.ndarray = np.array([])
+        self._index_by_drillhole: dict = {}
```

Comment:
Better to instantiate to None instead of mutable

Reply:
ok but it makes a lot of boring pylint and type gestion. 
With an empty iterable, when it has to be iterate, the loop pass and everyone's happy.